### ISSUE TITLE
FP needless-late-init: assignment inside brace expr

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9895]
### Summary

.

### Lint Name

needless-late-init

### Reproducer

I tried this code:

```rust
fn main() {
	let r;
	(r = 5);
}
```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::needless_late_init`
```
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected expression, found `let` statement
 --> src/main.rs:5:3
  |
5 |     (let r = 5);
  |      ^^^

error: `let` expressions are not supported here
 --> src/main.rs:5:3
  |
5 |     (let r = 5);
  |      ^^^^^^^^^
  |
  = note: only supported directly in conditions of `if` and `while` expressions
note: `let`s wrapped in parentheses are not supported in a context with let chains
 --> src/main.rs:5:3
  |
5 |     (let r = 5);
  |      ^^^^^^^^^

error[E0658]: `let` expressions in this position are unstable
 --> src/main.rs:5:3
  |
5 |     (let r = 5);
  |      ^^^^^^^^^
  |
  = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
  = help: add `#![feature(let_chains)]` to the crate attributes to enable

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0658`.
Original diagnostics will follow.

warning: variable `r` is assigned to, but never used
 --> src/main.rs:4:6
  |
4 |     let r;
  |         ^
  |
  = note: consider using `_r` instead
  = note: `#[warn(unused_variables)]` on by default

warning: value assigned to `r` is never read
 --> src/main.rs:5:3
  |
5 |     (r = 5);
  |      ^
  |
  = help: maybe it is overwritten before being read?
  = note: `#[warn(unused_assignments)]` on by default

warning: unneeded late initialization
 --> src/main.rs:4:2
  |
4 |     let r;
  |     ^^^^^^ created here
5 |     (r = 5);
  |     ^^^^^^^ initialised here
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_late_init
  = note: requested on the command line with `-W clippy::needless-late-init`
help: declare `r` here
  |
5 |     (let r = 5);
  |      ~~~~~

warning: `clpy` (bin "clpy") generated 3 warnings (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: `clpy` (bin "clpy" test) generated 3 warnings (3 duplicates)
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
```

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
